docs/API.md

```diff
@@ -28,15 +25,24 @@ class Dirty
 end
 END
 
-source_code = Reek::Source::SourceCode.from(source)
 reporter = Reek::CLI::Report::TextReport.new
-reporter.add_examiner Reek::Core::Examiner.new(source_code)
+examiner = Reek::Examiner.new(source)
+reporter.add_examiner examiner
+
 puts reporter.show
 ```
 
 This will show the list of errors in variable `source`.
 
-`reek` can take `source` as `String`, `File` or `IO`.
+`Reek::Examiner.new` can take `source` as `String`, `File` or `IO`, or an array
+of file names.
+
+```
+# Examine a file object
+reporter.add_examiner Reek::Examiner.new(File.new('dirty.rb'))
+# Examine source from an array of file names
+reporter.add_examiner Reek::Examiner.new(['also_dirty.rb', 'clean.rb')
+```
```

Comment:
One final thing: I'm not too sure about `report.examiners <<` versus `report.add_examiner`.

The naïve implementation of `report.examiners <<` would basically boil down to `attr_accessor :examiners`, which would expose the array of examiners to the outside world and allow any manipulation. An alternative would be to expose some proxy object that only allows `:<<`, but that would be confusing when people try `report.examiners = report.examiners + some_more_examiners`.

I don't think `add_examiner` is so bad.